operator/internal/manifests/build.go

```diff
@@ -98,12 +99,18 @@ func BuildAll(opts Options) ([]client.Object, error) {
 	if opts.Gates.LokiStackAlerts {
 		prometheusRuleObjs, err := BuildPrometheusRule(opts)
 		if err != nil {
-			return nil, err
+			return nil, lokiv1.NetworkPoliciesDisabled, err
 		}
 		res = append(res, prometheusRuleObjs...)
 	}
 
-	return res, nil
+	if opts.Stack.Tenants != nil && opts.FeatureGate.NetworkPoliciesEnabled(opts.Stack.Tenants.NetworkPolicies) {
+		networkPolicyObjs := BuildNetworkPolicies(opts)
+		res = append(res, networkPolicyObjs...)
+		networkPolicyStatus = lokiv1.NetworkPoliciesEnabled
+	}
```

Comment:
I think I would pull the "decision about NetworkPolicies" out of `BuildAll` and just have `opts` contain a boolean that decides whether to build NetworkPolicies or not. This should remove the need to have the `networkPolicyStatus` as a return value.